bin/check-es-index-field-count.rb

```diff
@@ -0,0 +1,197 @@
+#! /usr/bin/env ruby
+#
+#   check-es-index-field-count
+#
+# DESCRIPTION:
+#   This plugin checks if the number of fields in ES index(es) is approaching limit. ES by default
+#   puts this limit at 1000 fields per index.
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: aws_es_transport
+#
+# USAGE:
+#   This example checks if the number of fields for an index is reaching its limit set in
+#   index.mapping.total_fields.limit. This check takes as paramater the index name or
+#   comma-separated list of indices, and optionally the type to check on. If type is not specified,
+#   all types in index are examined.
+#   You can also specify an optional value for the limit. When omitted, this will default to 1000.
+#
+#   check-es-query-count.rb -h <hostname or ip> -p 9200
+#           -i <index1>,<index2> --types <type_in_index> -w <pct> -c <pct>
+#
+#   If any indices crossing the specified thresholds (warning/critical), beside the appropriate return code
+#   this check will also output a list of indices with the violated percentage for further troubleshooting.
+# NOTES:
+#
+# LICENSE:
+#   CloudCruiser <devops@hpe.com>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'elasticsearch'
+require 'sensu-plugins-elasticsearch'
+require 'json'
+
+#
+# ES Index Field Count
+#
+class ESIndexFieldCount < Sensu::Plugin::Check::CLI
+  include ElasticsearchCommon
+
+  option :host,
+         description: 'Elasticsearch host',
+         short: '-h HOST',
+         long: '--host HOST',
+         default: 'localhost'
+
+  option :port,
+         description: 'Elasticsearch port',
+         short: '-p PORT',
+         long: '--port PORT',
+         proc: proc(&:to_i),
+         default: 9200
+
+  option :scheme,
+         description: 'Elasticsearch connection scheme, defaults to https for authenticated connections',
+         short: '-s SCHEME',
+         long: '--scheme SCHEME'
+
+  option :password,
+         description: 'Elasticsearch connection password',
+         short: '-P PASSWORD',
+         long: '--password PASSWORD'
+
+  option :user,
+         description: 'Elasticsearch connection user',
+         short: '-u USER',
+         long: '--user USER'
+
+  option :timeout,
+         description: 'Elasticsearch query timeout in seconds',
+         short: '-t TIMEOUT',
+         long: '--timeout TIMEOUT',
+         proc: proc(&:to_i),
+         default: 30
+
+  option :index,
+         description: 'Elasticsearch indices to check against.
+         Comma-separated list of index names to search.
+         Use `_all` or empty string to perform the operation on all indices.
+         Accepts wildcards',
+         short: '-i INDEX',
+         long: '--indices INDEX'
```

Comment:
I think we should default to `_all` as I imagine most people want to check all indices.